awswrangler/lakeformation/_utils.py

```diff
@@ -1,18 +1,134 @@
 """Utilities Module for Amazon Lake Formation."""
 import logging
 import time
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, List, Optional, Union
 
 import boto3
 
 from awswrangler import _utils, exceptions
+from awswrangler.catalog._utils import _catalog_id
+from awswrangler.s3._describe import describe_objects
 
 _QUERY_FINAL_STATES: List[str] = ["ERROR", "FINISHED"]
 _QUERY_WAIT_POLLING_DELAY: float = 2  # SECONDS
 
 _logger: logging.Logger = logging.getLogger(__name__)
 
 
+def _without_keys(d: Dict[str, Any], keys: List[str]) -> Dict[str, Any]:
+    return {x: d[x] for x in d if x not in keys}
+
+
+def _build_partition_predicate(
+    partition_cols: List[str],
+    partitions_types: Dict[str, str],
+    partitions_values: List[str],
+) -> str:
+    partition_predicates: List[str] = []
+    for col, val in zip(partition_cols, partitions_values):
+        if partitions_types[col].startswith(("tinyint", "smallint", "int", "bigint", "float", "double", "decimal")):
+            partition_predicates.append(f"{col}={str(val)}")
+        else:
+            partition_predicates.append(f"{col}='{str(val)}'")
+    return " AND ".join(partition_predicates)
```

Comment:
Return partition_predicate based on received partitions. E.g. `name="elizabeth" AND age=91`. The syntax depends on the type of the partitions